torch/csrc/autograd/FunctionsManual.cpp

```diff
@@ -622,6 +622,42 @@ Tensor _sparse_addmm_sparse_backward(const Tensor& grad, const Tensor& sparse_,
   return grad_sparse.sparse_mask(sparse);
 }
 
+template <short grad_order>
+Tensor sparse_matmul_kernel_grad(const Tensor& grad_, const Tensor& x) {
+  Tensor grad = fill_with_ones(grad_.coalesce());
```

Comment:
@ngimel and I don't think we need to fill the gradient with ones.

In particular, it seems like there should be a way to do backward with sparse @ sparse matmul, and not materialize a strided tensor here. The unspecified values of the gradient won't affect the masked gradients of the inputs.

Reply:
Ohh Thanks! This weekend I was trying to figure out something in that direction. Based on your feedback  I came out with the following:  

```python
def (a, b, c_grad):
	a_grad = torch.sparse.mm(c_grad, b.t())
	b_grad = torch.sparse.mm(a.t(), c_grad)

	a_grad = sparse_matrix_mask(a_grad.coalesce(), a.coalesce())
	b_grad = sparse_matrix_mask(b_grad.coalesce(), b.coalesce())
```
and it works. Right now I am working implementing an efficient function `sparse_matrix_mask` which is basically a `sparse_mask` function but for 2-d sparse tensors.